### PR TITLE
feat: map container environment variables to chart

### PR DESCRIPTION
**Pull request by d7oc**
This PR is aimed to add all environment variables from docker to the helm chart.

Current TODO:

- [x] Decide how empty values should be handled.
- [x] Fix the few occurring docker environment defaults where the documentation is missing.
- [x] Decide if cron setting should be split.
- [x] Decide if memcache setting should be split
- [ ] Add secrets usage for sensitive variables

### CODE REVIEW

File: charts/owncloud/templates/deployment.yaml [hunk 1]
```diff
@@ -48,61 +48,364 @@ spec:
...
+            - name: OWNCLOUD_ALLOW_USER_TO_CHANGE_DISPLAY_NAME
+              value: {{ .Values.owncloud.allowUserToChangeDisplayName | quote }}
+            - name: OWNCLOUD_APPS_DEPRECATED
+              value: {{ .Values.owncloud.apps.deprecated | quote }}
+            - name: OWNCLOUD_APPS_DISABLE
+              value: {{ .Values.owncloud.apps.disable | quote }}
+            - name: OWNCLOUD_APPS_ENABLE
+              value: {{ .Values.owncloud.apps.enable | quote }}
+            - name: OWNCLOUD_APPS_INSTALL
+              value: {{ .Values.owncloud.apps.enstall | quote }}
```
Comment (xoxys) [inline charts/owncloud/templates/deployment.yaml]: apps.**i**nstall

File: charts/owncloud/templates/deployment.yaml [hunk 2]
```diff
@@ -48,61 +48,364 @@ spec:
...
+            - name: OWNCLOUD_MAX_EXECUTION_TIME
+              value: {{ .Values.owncloud.maxExecutionTime | quote }}
+            - name: OWNCLOUD_MAX_FILESIZE_ANIMATED_GIFS_PUBLIC_SHARING
+              value: {{ .Values.owncloud.maxFilesizeAnimatedGifsPublicSharing | quote }}
+            - name: OWNCLOUD_MAX_INPUT_TIME
+              value: {{ .Values.owncloud.maxInputTime | quote }}
+            - name: OWNCLOUD_MAX_UPLOAD
+              value: {{ .Values.owncloud.maxUpload | quote }}
+            - name: OWNCLOUD_MEMCACHED_ENABLED
+              value: {{ .Values.owncloud.memcachedEnabled | quote }}
```
Comment (xoxys) [inline charts/owncloud/templates/deployment.yaml:226]: Memcached options namespacing?  `.Values.owncloud.memcached.enabled`

File: charts/owncloud/values.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ image:
   # -- Image repository
   repository: docker.io/owncloud/server
   # -- Image tag.
-  # @default -- The `appVersion` of the Chart.
+    # @default -- The `appVersion` of the Chart.
```
Comment (xoxys) [inline charts/owncloud/values.yaml]: Wrong indention.

File: charts/owncloud/values.yaml [hunk 4]
```diff
@@ -17,29 +17,543 @@ nameOverride: ""
...
 owncloud:
-  domain: owncloud.chart.example
-  username: owncloud
-  password: owncloud
-  volume_apps: /mnt/data/apps
-  volume_config: /mnt/data/config
-  volume_files: /mnt/data/files
-  volume_root: /mnt/data
+  # -- Location of the access log.
+  # @default -- /dev/stdout
```
Comment (xoxys) [inline charts/owncloud/values.yaml:21]: The default is set automatically from the value. You just need to override it in some special cases with an `@default` line. Can be removed in most cases below as well.

File: charts/owncloud/values.yaml [hunk 5]
```diff
@@ -17,29 +17,543 @@ nameOverride: ""
...
+  # -- Define MySQL 3/4 byte character handling (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/server/config_sample_php_parameters.html#define-mysql-34-byte-character-handling)).
+  # @default -- ${OWNCLOUD_UTF8MB4_ENABLED}
+  mysqlUtf8Mb4: ""
+  objectstore:
+    # -- Bucket name to store data.
+    # @default -- owncloud
+    bucket: "owncloud"
+    # -- Class to use for the objectstore. It is recommended to keep the default (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/files/external_storage/s3_compatible_object_storage_as_primary.html)).
+    # @default -- OCA\Files_Primary_S3\S3Storage
+    class: "OCA\Files_Primary_S3\S3Storage"
```
Comment (xoxys) [inline charts/owncloud/values.yaml]: YAML needs `\\` to escape or single quotes around the entire value.

**Review by xoxys — CHANGES_REQUESTED** [review]

File: charts/owncloud/templates/deployment.yaml (hunk 1 above)
Comment (d7oc) [inline charts/owncloud/templates/deployment.yaml]:
```suggestion
              value: {{ .Values.owncloud.apps.install | quote }}
```

Comment (d7oc) [reply charts/owncloud/templates/deployment.yaml]: fixed

File: charts/owncloud/templates/deployment.yaml (hunk 2 above)
Comment (d7oc) [reply charts/owncloud/templates/deployment.yaml:226]: Yes that's exactly why this is on the TODO list above. I'm not sure if we should split it, as some settings start with `memcached` and some with `memcache` without `d`. If we switch to name spacing we lose the direct transferability from the docker name.

File: charts/owncloud/values.yaml (hunk 3 above)
Comment (d7oc) [inline charts/owncloud/values.yaml]:
```suggestion
  # @default -- The `appVersion` of the Chart.
```

File: charts/owncloud/values.yaml (hunk 4 above)
Comment (d7oc) [reply charts/owncloud/values.yaml:21]: Fixed

File: charts/owncloud/values.yaml (hunk 3 above)
Comment (d7oc) [reply charts/owncloud/values.yaml]: fixed

File: charts/owncloud/values.yaml (hunk 5 above)
Comment (d7oc) [reply charts/owncloud/values.yaml]:
```suggestion
    class: "OCA\\Files_Primary_S3\\S3Storage"
```

File: charts/owncloud/values.yaml [hunk 6]
```diff
@@ -17,29 +17,386 @@ nameOverride: ""
...
+  adminUsername: "owncloud"
+  # -- Allow or disallow users to change their display names (see [documentation](https://doc.owncloud.com/server/latest/admin_manual/configuration/server/config_sample_php_parameters.html#allow-or-disallow-users-to-change-their-display-names)).
+  allowUserToChangeDisplayName: ""
+  apps: 
+    # -- List of deprecated apps that must be removed (automatically) before performing an ownCloud upgrade to avoid upgrade issues.
+    deprecated: ""
+    # -- List of apps to disable on container startup.
+    disable: ""
+    # -- List of apps to enable on container startup.
+    # @default -- {{ .Values.owncloud.apps.install }}
```
Comment (xoxys) [inline charts/owncloud/values.yaml]: Would you mind adding code fences to all of those values? E.g. `` `{{ .Values.owncloud.apps.install }}` ``

**Review by xoxys — CHANGES_REQUESTED** [review]
Just a small one, LGTM otherwise.

**Review by xoxys — APPROVED** [review]

File: charts/owncloud/values.yaml (hunk 6 above)
Comment (d7oc) [reply charts/owncloud/values.yaml]: Done, see https://github.com/owncloud-docker/helm-charts/pull/11/commits/cd0b5075616018ffe880a3c423c087ee189671bc and https://github.com/owncloud-docker/helm-charts/pull/11/commits/1d4239b5250756b6e2873bf90a9d770ce9baab42

**Pull request merged**